### DISCUSSION TITLE
Available option

### DISCUSSION THREAD

**Discussion by DonatoD** [gapitio/gapit-htmlgraphics-panel#95] [Q&A] [answered]
Hi, is it possible to have some more details on the available options? (Dynamic htmlGraphics, Dynamic htmlGraphics, etc)
The short description is not so clear. Even after spending quite a lot of time trying on several code, actually ist's still unclear to me.
Thanks

**Comment by ZuperZee** [comment, marked as answer]
Thanks for bringing this up, I'll try to update the docs to be more clear.

There is a thing in docs about [dynamic variables](https://gapit-htmlgraphics-panel.gapit.io/docs/advanced/dynamic-variables/), but might not be what's unclear.

Here is a link to all the options _(there are some links to which variable/reference it updates, like [Dynamic data](https://gapit-htmlgraphics-panel.gapit.io/docs/options/#dynamic-data))_ https://gapit-htmlgraphics-panel.gapit.io/docs/options

A bit more information about *Dynamic htmlGraphics* (and the other dynamic stuff) and when to use it (I'll add it to docs later).

When *Dynamic htmlGraphics* is true
![image](https://user-images.githubusercontent.com/39485579/158678649-65c67594-340e-4b93-98bb-52c1a2941b08.png)
it updates the [htmlGraphics object](https://gapit-htmlgraphics-panel.gapit.io/docs/references/#htmlgraphics-global) when new data is available. new data mostly happens when a refresh happens.
![image](https://user-images.githubusercontent.com/39485579/158682261-d8a2b759-d689-481b-9ba9-ff75aa4f84dc.png)
The console output:  ![image](https://user-images.githubusercontent.com/39485579/158682303-768a50a4-04f6-41ff-b2e4-a7e4241a94d0.png)

A simple example where dynamic data object is useful.
![dynamic](https://user-images.githubusercontent.com/39485579/158687064-ff4a549b-3e9d-4bef-9a0e-778e61b255f5.gif)
Both panels have the same code, except that [Dynamic data object](https://gapit-htmlgraphics-panel.gapit.io/docs/options/#dynamic-data) is false on the top panel and true on the bottom panel.

Panel json (Dynamic data object = false)
```json
{
    "add100Percentage": true,
    "centerAlignContent": true,
    "overflow": "visible",
    "SVGBaseFix": true,
    "codeData": "{\n  \"text\": \"Random text\"\n}",
    "rootCSS": "",
    "css": "* {\n  font-family: Open Sans;\n}\n",
    "html": "<div>\n  <div>Actual value: <strong id=\"actual-value\"></strong></div>\n  <div>onInit Value: <strong id=\"value\"></strong></div>\n</div>\n",
    "renderOnMount": true,
    "onRender": "const actualValueElt = htmlNode.getElementById('actual-value');\n\nconst valueField = data.series[0]?.fields[1];\nconst length = valueField.values.length;\nconst value = valueField.values.get(length - 1)\n\nactualValueElt.textContent = value;\n",
    "dynamicHtmlGraphics": true,
    "dynamicData": false,
    "dynamicFieldDisplayValues": false,
    "dynamicProps": false,
    "panelupdateOnMount": true,
    "onInitOnResize": false,
    "onInit": "const valueElt = htmlNode.getElementById('value');\n\nfunction update() {\n  const valueField = data.series[0]?.fields[1];\n  const length = valueField.values.length;\n  const value = valueField.values.get(length - 1)\n  valueElt.textContent = value;\n}\n\nhtmlNode.addEventListener(\"panelupdate\", update);\n"
}
```

Panel json (Dynamic data object = true)
```json
{
    "add100Percentage": true,
    "centerAlignContent": true,
    "overflow": "visible",
    "SVGBaseFix": true,
    "codeData": "{\n  \"text\": \"Random text\"\n}",
    "rootCSS": "",
    "css": "* {\n  font-family: Open Sans;\n}\n",
    "html": "<div>\n  <div>Actual value: <strong id=\"actual-value\"></strong></div>\n  <div>onInit Value: <strong id=\"value\"></strong></div>\n</div>\n",
    "renderOnMount": true,
    "onRender": "const actualValueElt = htmlNode.getElementById('actual-value');\n\nconst valueField = data.series[0]?.fields[1];\nconst length = valueField.values.length;\nconst value = valueField.values.get(length - 1)\n\nactualValueElt.textContent = value;\n",
    "dynamicHtmlGraphics": true,
    "dynamicData": false,
    "dynamicFieldDisplayValues": false,
    "dynamicProps": false,
    "panelupdateOnMount": true,
    "onInitOnResize": false,
    "onInit": "const valueElt = htmlNode.getElementById('value');\n\nfunction update() {\n  const valueField = data.series[0]?.fields[1];\n  const length = valueField.values.length;\n  const value = valueField.values.get(length - 1)\n  valueElt.textContent = value;\n}\n\nhtmlNode.addEventListener(\"panelupdate\", update);\n"
}
```

In general the dynamic stuff is only useful if it's used in an event like [panelupdate](https://gapit-htmlgraphics-panel.gapit.io/docs/options/#dynamic-data), [onclick](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onclick), onmouseenter ETC.

Please let me know if this helped :D

**Reply by DonatoD** [reply]
Thanks so much

**Reply by ZuperZee** [reply]
Just curious, what made it easier to understand?

**Reply by DonatoD** [reply]
Honestly I didn't had time to have look at it in detail, but I see you even indicate some more links with more instruction. For sure they will help me more
